### ISSUE TITLE
Export not working for large backups

### ISSUE THREAD

**Issue by arjen-w** [jcwillox/hass-auto-backup#17]
I have configured /Export as export folder for the alternate backup location. This is working fine for small full backups (1Gb). Now my backup is about 2Gb and the export file is 0kb.
The rror in the server log:

```
Traceback (most recent call last):
  File "/config/custom_components/auto_backup/__init__.py", line 339, in new_snapshot
    await self.copy_snapshot(data[ATTR_NAME], slug, backup_path)
  File "/config/custom_components/auto_backup/__init__.py", line 432, in copy_snapshot
    await self.download_snapshot(slug, destination)
  File "/config/custom_components/auto_backup/__init__.py", line 452, in download_snapshot
    file.write(await request.read())
  File "/usr/local/lib/python3.7/site-packages/aiohttp/client_reqrep.py", line 973, in read
    self._body = await self.content.read()
  File "/usr/local/lib/python3.7/site-packages/aiohttp/streams.py", line 362, in read
    return b''.join(blocks)
MemoryError
```

Is there a way that this can be fixed? I am using an RPI4 with 4 Gb of memory. Disk space is plenty available.

**Comment by jcwillox** [comment]
Well, currently the download method actually loads the file into memory before writing it to disk so that'd be why you getting the memory error with the larger file size. Even 1GB is a huge amount of memory for one process to be using.
It seems that this is fixable, just needs to be a chunked stream.

Side question where is the `/export` folder, I haven't had any luck downloading backups to locations other than `/config` and `/share` (mainly because the other folders are not mounted to the homeassistant docker container).

**Comment by arjen-w** [comment]
The Export folder is located in /usr/share/hassio/homeassistant/ I mainly use the tool to rename the backup files to contain the date and time of the snapshot in the file name. After that, I run a scheduled script to delete files older than 2 days and copy the newer files to my NAS.

**Comment by jcwillox** [comment]
Thanks, I thought it would be something like that.
Anyway, this should be fixed in the latest release ([0.5.2](https://github.com/jcwillox/hass-auto-backup/releases/tag/0.5.2)), feel free to re-open if it's not.

**Comment by arjen-w** [comment]
Issue re-occurred, but with a different error:
2020-05-12 00:38:03 ERROR (MainThread) [custom_components.auto_backup] Client error on /snapshots/c470b479/download request Response payload is not completed
2020-05-12 00:38:03 ERROR (MainThread) [custom_components.auto_backup] Error during backup. Snapshot download failed. Check the logs for more information.
2020-05-12 00:38:03 ERROR (MainThread) [homeassistant.core] Error doing job: Unclosed connection

**Comment by jcwillox** [comment]
That's quite an odd, does this happen every time?
I'm honestly not sure why this would be happening, it not a timeout on the client end, maybe on the supervisor? It also doesn't appear to be file or storage error.

Researching that error, it can only happen if the response object is closed before we read all the data from it, or if the server doesn't send as much data as it said it would. Basically, some kind of network error appears to be occurring that results in the client not receiving all of the data.

**Comment by arjen-w** [comment]
I happende the last 2 days. It was running fine at the 10th of May, but failing yesterday and this morning. All actions are running on the local machine, so I can't see why this would be a connection error. Both file were larger than 2Gb, while the file that was succesful was 1.9Gb. It can therefore be a sizing issue? I am lowering the amount of devices that are logging to Influxdb, so the backup size should shrink in the next couple of days, if that would be the issue.

**Comment by jcwillox** [comment]
Yeah sounds like its an issue with the size. I don't believe data is ever deleted from influxdb so that would only change the rate of increase but your backups will still be increasing in size.

You may want to try to reduce the size, for reference with 315 entities (133 sensors) my full backups are basically constant at 95 MB, to reach that though many entities are excluded from the database, currently I store states for 107 entities, and I use a custom longterm database that only stores the timestamp and state for select sensors (and some other tools). This is a far cry from when I unknowingly let by HA database get to 10 GB 😄

**Comment by arjen-w** [comment]
File is around 2Gb now and export is working fine now. Seems indeed a file size issue.
FYI: I changed the retention period for InfluxDB to 2 weeks, so database will shrink more. I will also look at the information that will be in the HA db itself as that one is now about 1.5Gb.

**Comment by arjen-w** [comment]
I can confirm now it is due to large file.
Turns out the Export folder (where I stored the backups) was also part of the backup. I now shrink the Home assistant db by running an auto pure task each night and I am emptying the export folder once the export has been moved to another folder, so my backup is now very small and the extension is working fine. So as long as the file is not larger than 2Gb, there is no issue.
